library/lua/time.lua

```diff
@@ -0,0 +1,87 @@
+local _ENV = mkmodule('time')
+
+local TU_PER_DAY = 1200
+--[[
+if advmode then TU_PER_DAY = 86400 ? or only for cur_year_tick?
+advmod_TU / 72 = ticks
+--]]
+local TU_PER_MONTH = TU_PER_DAY * 28
+local TU_PER_YEAR = TU_PER_MONTH * 12
+
+local MONTHS = {
+ 'Granite',
+ 'Slate',
+ 'Felsite',
+ 'Hematite',
+ 'Malachite',
+ 'Galena',
+ 'Limestone',
+ 'Sandstone',
+ 'Timber',
+ 'Moonstone',
+ 'Opal',
+ 'Obsidian',
+}
+
+Time = defclass(Time)
+function Time:init(args)
+ self.year = args.year or 0
+ self.ticks = args.ticks or 0
+end
+function Time:getDays() -- >>float<< Days as age (including years)
+ return self.year * 336 + (self.ticks / TU_PER_DAY)
+end
+function Time:getDayInYear() -- >>int<< Current day of the year
+ return math.floor (self.ticks / TU_PER_DAY)
+end
+function Time:getDayInMonth()
+ return math.floor ( (self.ticks % TU_PER_MONTH) / TU_PER_DAY ) + 1
+end
+function Time:getMonths() -- >>int<< Months as age (not including years)
+ return math.floor (self.ticks / TU_PER_MONTH)
+end
+function Time:getYears() -- >>int<<
+ return self.year
+end
+function Time:getMonthStr() -- Month as date
+ return MONTHS[self:getMonths()+1] or 'error'
+end
+function Time:getDayStr() -- Day as date
+ local d = math.floor ( (self.ticks % TU_PER_MONTH) / TU_PER_DAY ) + 1
+ if d == 11 or d == 12 or d == 13 then
+  d = tostring(d)..'th'
+ elseif d % 10 == 1 then
+  d = tostring(d)..'st'
+ elseif d % 10 == 2 then
+  d = tostring(d)..'nd'
+ elseif d % 10 == 3 then
+  d = tostring(d)..'rd'
+ else
+  d = tostring(d)..'th'
+ end
+ return d
```

Comment:
type changing confuses the linter. It would be clearer if you return the value in each of those branches instead of setting `d` and returning at the end